src/media/ros/ros_reader.cpp

```diff
@@ -948,19 +948,48 @@ namespace librealsense
         {
             sensor_extensions[RS2_EXTENSION_COLOR_SENSOR] = std::make_shared<color_sensor_snapshot>();
         }
-        if (is_motion_module_sensor(sensor_name))
+        else if( is_motion_module_sensor( sensor_name ) )
         {
             sensor_extensions[RS2_EXTENSION_MOTION_SENSOR] = std::make_shared<motion_sensor_snapshot>();
         }
-        if (is_fisheye_module_sensor(sensor_name))
+        else if( is_fisheye_module_sensor( sensor_name ) )
         {
             sensor_extensions[RS2_EXTENSION_FISHEYE_SENSOR] = std::make_shared<fisheye_sensor_snapshot>();
         }
+        else if( is_depth_sensor( sensor_name ) )
+        {
+            if( sensor_extensions.find( RS2_EXTENSION_DEPTH_SENSOR ) == nullptr )
+            {
+                float depth_units = 0.01f; // Default to 1mm for devices that don't have this option implemented
+                sensor_extensions[RS2_EXTENSION_DEPTH_SENSOR] = std::make_shared< depth_sensor_snapshot >( depth_units );
+
+                if( is_stereo_depth_sensor( sensor_name ) ) // Need both extensions
+                {
+                    if( sensor_extensions.find( RS2_EXTENSION_DEPTH_STEREO_SENSOR ) == nullptr )
+                    {
+                        float baseline = 0.095f; // Default for D555 (and D455 but D400 have baseline option implemented and won't need this)
+                        for( auto & ext : m_extrinsics_map ) // Get real value from extrinsics data, if exists
+                        {
+                            if( ext.first.stream_type == RS2_STREAM_INFRARED && ext.first.stream_index == 2 )
+                                baseline = ext.second.second.translation[0];
+                        }
+                        sensor_extensions[RS2_EXTENSION_DEPTH_STEREO_SENSOR] = std::make_shared< depth_stereo_sensor_snapshot >( depth_units, baseline );
+                    }
+                }
+            }
+        }
     }
 
     bool ros_reader::is_depth_sensor(std::string sensor_name)
     {
-        if (sensor_name.compare("Stereo Module") == 0 || sensor_name.compare("Coded-Light Depth Sensor") == 0)
+        if( is_stereo_depth_sensor( sensor_name ) || sensor_name.compare( "Coded-Light Depth Sensor" ) == 0 )
```

Comment:
can be rplacced by:
return ( is_stereo_depth_sensor( sensor_name ) || sensor_name.compare( "Coded-Light Depth Sensor" ) == 0 );
Instead of the return true, return false.
Can also be inlined in the h file (as the implementation is so short.